openmrs/omod/src/main/java/org/projectbuendia/openmrs/web/controller/ProfileManager.java

```diff
@@ -48,34 +42,68 @@
 /** The controller for the profile management page. */
 @Controller
 public class ProfileManager {
-    static Log log = LogFactory.getLog(ProfileManager.class);
-    final File PROFILE_DIR = new File("/usr/share/buendia/profiles");
-    final String VALIDATE_CMD = "buendia-profile-validate";
-    final String APPLY_CMD = "buendia-profile-apply";
+    protected static Log log = LogFactory.getLog(ProfileManager.class);
+    private final String PROFILE_DIR_PATH = "/usr/share/buendia/profiles";
+    private final String VALIDATE_CMD = "buendia-profile-validate";
+    private final String APPLY_CMD = "buendia-profile-apply";
+    private File profileDir;
+
+    /** This is executed every time a request is made and determines the profileDir to be used. */
+    @ModelAttribute
+    public void getProfileDir() {
+        Properties config = Context.getRuntimeProperties();
+        String configProfileDir = config.getProperty("profile_manager.profile_dir", PROFILE_DIR_PATH);
+        profileDir = new File(configProfileDir);
+    }
 
     @RequestMapping(value = "/module/projectbuendia/openmrs/profiles", method = RequestMethod.GET)
     public void get(HttpServletRequest request, ModelMap model) {
-        List<FileInfo> files = new ArrayList<>();
-        for (File file : PROFILE_DIR.listFiles()) {
-            files.add(new FileInfo(file));
+        String currentProfile = Context.getAdministrationService().getGlobalProperty(GlobalProperties.CURRENT_PROFILE);
+        model.addAttribute("currentProfile", currentProfile);
+        model.addAttribute("authorized", authorized());
+        model = queryParamsToModelAttr(request, model);
```

Comment:
What does this snippet does ? It would be more clear if you comment it.
```
        String currentProfile = Context.getAdministrationService().getGlobalProperty(GlobalProperties.CURRENT_PROFILE);
        model.addAttribute("currentProfile", currentProfile);
        model.addAttribute("authorized", authorized());
        model = queryParamsToModelAttr(request, model);
```